File: frontend/src/types/contract.ts
import type { TravelResponse } from "./travel";

export interface Passenger {
  id?: string;
  name: string;
  cpf: string;
  rg: string;
  birthDate: string;
}

export interface ContractRequest {
  clientName: string;
  clientCpf: string;
  clientRg: string;
  clientBirthDate: string;
  clientPhone: string;
  addressStreet: string;
  addressNumber: string;
  addressComplement?: string;
  addressNeighborhood: string;
  addressCity: string;
  addressState: string;
  addressZip: string;
  travelId: string;
  passengers: Passenger[];
}

export interface ContractResponse extends Omit<ContractRequest, 'travelId'> {
  id: string;
  status: "PENDING" | "APPROVED" | "PAID" | "CONFIRMED" | "CANCELLED";
  priceTotal: number;
  paymentMethod: string;
  roomType: string;
  tokenAccess?: string;
  travel: TravelResponse;
}

export interface PassengerListDTO {
    number: number;
    name: string;
    cpf: string;
    birthDate: string;
    age: string;
    roomType: string;
}

File: backend/src/main/java/com/agencia/viagens/model/Contract.java
package com.agencia.viagens.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Contract {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "travel_id")
    private Travel travel;

    private String clientName;
    private String clientCpf;
    private String clientRg;
    private String clientBirthDate;
    private String clientPhone;

    private String addressStreet;
    private String addressNumber;
    private String addressComplement;
    private String addressNeighborhood;
    private String addressCity;
    private String addressState;
    private String addressZip;

    @OneToMany(mappedBy = "contract", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Passenger> passengers;

    @OneToMany(mappedBy = "contract", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<ClientPayment> clientPayments;

    private Integer totalPeople;
    private BigDecimal priceTotal;
    private String paymentMethod;
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    private ContractStatus status;

    @Column(unique = true, nullable = false)
    private UUID tokenAccess;

    private String roomType;

    @OneToMany(mappedBy = "contract", cascade = CascadeType.ALL)
    private List<Payment> payments = new ArrayList<>();
}
